test/e2e_node/oomkiller_test.go

```diff
@@ -122,3 +154,26 @@ func getOOMTargetContainer(name string) v1.Container {
 		},
 	}
 }
+
+// getOOMTargetContainerMultiProcess returns a container with two processes, one of which attempts to allocate more
+// memory than is allowed by the container memory limit, and a second process which just sleeps
+func getOOMTargetContainerMultiProcess(name string) v1.Container {
```

Comment:
qq: Looks to me that without setting memory.oom.group, multi process container still has failed and 137 as exit code. I'm wondering will the new added test cover new added oom.group feature?

Reply:
Sorry, I don't understand this comment.  Without setting the new cgroup flag, individual processes are killed in the container, but the container is still running.  You can verify this by running this pod:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  containers:
  - name: bash
    resources:
      limits:
        memory: 10M
    image: busybox:latest
    command: ["sh","-c", "dd if=/dev/zero of=/dev/null bs=20M & sleep 86400"]
```

On my cluster, the pod and container still look ok:

```bash
$ k get pod test-pod -o yaml | yq '.status'
conditions:
  - lastProbeTime: null
    lastTransitionTime: "2023-05-09T18:07:07Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2023-05-09T18:07:09Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2023-05-09T18:07:09Z"
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2023-05-09T18:07:07Z"
    status: "True"
    type: PodScheduled
containerStatuses:
  - containerID: containerd://7fd5eda37230c4a532632ca19da29e8a72c0f1d2d05e7f3059e783dac4c656f0
    image: docker.io/library/busybox:latest
    imageID: docker.io/library/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16
    lastState: {}
    name: bash
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2023-05-09T18:07:09Z"
hostIP: 192.168.67.135
phase: Running
podIP: 192.168.93.235
podIPs:
  - ip: 192.168.93.235
qosClass: Burstable
startTime: "2023-05-09T18:07:07Z"
```

But on my node, I can see that dd was killed:
```
May  9 18:07:09 i-00706ac6ebbe246c9 kernel: dd invoked oom-killer: gfp_mask=0xcc0(GFP_KERNEL), order=0, oom_score_adj=999
```
```
10187 ?        Sl     0:00 /usr/bin/containerd-shim-runc-v2 -namespace k8s.io -id d7e43a3f4dac52075c3ebcdb88f375ad134a2dfca49f45bab65d7a9cb7b5357e -address /run/containerd/containerd.sock
10210 ?        Ss     0:00  \_ /pause
10244 ?        Ss     0:00  \_ sh -c dd if=/dev/zero of=/dev/null bs=20M & sleep 86400
10257 ?        Z      0:00      \_ [dd] <defunct>
```

Reply:
Thanks for sharing the info. I just reproduced the same result. Correct me if I'm wrong, in this test, without kill other processes(sleep) in the same cgroup, test supposed to failed because the timeout.

My apology made confusion in previous comments, because I tried similar test case last week(`["sh","-c", "dd if=/dev/zero of=/dev/null bs=20M & (sleep 100; exit 0)"]`) and I misunderstand the test.